pkg/controller/pxc/controller.go

```diff
@@ -186,13 +186,26 @@ func (r *ReconcilePerconaXtraDBCluster) Reconcile(request reconcile.Request) (re
 		return reconcile.Result{}, err
 	}
 
+	pxcService := pxc.NewServicePXC(o)
```

Comment:
looks like we overcomplicated solution, we run the same function for PXC and ProxySQL services and inside the function, we detect which Service we update via `ClusterIP != "None"`
I believe we 1) need different functions for PXC and ProxySQL 2) maybe we don't need to touch PXC Service at all

Reply:
Protocol: Since we don't mess with protocol on update it stays the same during the update

Service: I will leave only ProxySQL service update